### PR TITLE
Testing new tiers for RHODS workshops

### PR DESCRIPTION
**Pull request by cfchase**
Paired with https://github.com/codeready-toolchain/toolchain-e2e/pull/611

### CODE REVIEW

**Comment by cfchase** [comment]
@alexeykazakov The new space stuff is causing issues on my old test clusters, so I was having a hard time testing this separately.  Perhaps it's better if I can get this test tier used on stage and test it out there.  If this PR looks ok, I'll continue to test.

The idea here is to give 3GB of ram for default apps and 13GB overall to allow 3 running pods and 1 extra pod that can spin up as part of a normal rollout.  In addition, I created a separate build quota.

File: deploy/templates/nstemplatetiers/test/cluster.yaml [hunk 1]
```diff
@@ -6,15 +6,38 @@ objects:
 - apiVersion: quota.openshift.io/v1
   kind: ClusterResourceQuota
   metadata:
-    name: for-${USERNAME}-compute
+    name: for-${USERNAME}-compute-deploy
   spec:
     quota:
+      scopes:
+        - NotTerminating
```
Comment (alexeykazakov) [inline deploy/templates/nstemplatetiers/test/cluster.yaml:13]: AFAIK scopes are supported in `ResoruceQuotas` only. They are not supported in `ClusterResourceQuotas`

Comment (alexeykazakov) [reply deploy/templates/nstemplatetiers/test/cluster.yaml:13]: This is why we moved the compute and storage quotas to the `ns_*.yaml` from the `cluster.yaml` in our AppStudio tier: https://github.com/codeready-toolchain/host-operator/blob/master/deploy/templates/nstemplatetiers/appstudio/ns_appstudio.yaml#L57-L93

Comment (cfchase) [reply deploy/templates/nstemplatetiers/test/cluster.yaml:13]: Thanks.  Will do.

File: deploy/templates/nstemplatetiers/test/cluster.yaml [hunk 2]
```diff
@@ -128,15 +109,16 @@ objects:
...
   kind: Idler
   metadata:
-    name: ${USERNAME}-dev
-  spec:
-    timeoutSeconds: 43200 # 12 hours
-- apiVersion: toolchain.dev.openshift.com/v1alpha1
-  kind: Idler
-  metadata:
-    name: ${USERNAME}-stage
+    name: ${USERNAME}
```
Comment (cfchase) [inline deploy/templates/nstemplatetiers/test/cluster.yaml:112]: wasn't sure if you wanted this to be `${USERNAME}-dev` or ${USERNAME}`  now that it's single namespace

File: deploy/templates/nstemplatetiers/test/cluster.yaml [hunk 3]
```diff
@@ -3,25 +3,6 @@ kind: Template
 metadata:
   name: test-cluster-resources
 objects:
-- apiVersion: quota.openshift.io/v1
```
Comment (cfchase) [inline deploy/templates/nstemplatetiers/test/cluster.yaml:6]: Moved resource quotas to the namespace, so this seems unnecessary, but I will restore and set to cpu/memory that accomodates the resourcequota from the namespace if you prefer

File: deploy/templates/nstemplatetiers/test/cluster.yaml (hunk 2 above)
Comment (alexeykazakov) [reply deploy/templates/nstemplatetiers/test/cluster.yaml:112]: `${USERNAME}-dev` as it in our base1n tier.

Comment (cfchase) [reply deploy/templates/nstemplatetiers/test/cluster.yaml:112]: done

File: deploy/templates/nstemplatetiers/test/ns_dev.yaml [hunk 4]
```diff
@@ -13,6 +13,61 @@ objects:
...
+- apiVersion: v1
+  kind: ResourceQuota
+  metadata:
+    name: compute-deploy
+    namespace: ${USERNAME}-dev
+  spec:
+    scopes:
+      - NotTerminating
+    hard:
+      limits.cpu: "20"
```
Comment (alexeykazakov) [inline deploy/templates/nstemplatetiers/test/ns_dev.yaml:28]:
```suggestion
      limits.cpu: 20000m
```
It's the same but it's easier to test in e2e tests if it's defined in milli-cores (easier to compare strings in m).

File: deploy/templates/nstemplatetiers/test/ns_dev.yaml [hunk 5]
```diff
@@ -13,6 +13,61 @@ objects:
...
+- apiVersion: v1
+  kind: ResourceQuota
+  metadata:
+    name: compute-build
+    namespace: ${USERNAME}-dev
+  spec:
+    scopes:
+      - Terminating
+    hard:
+      limits.cpu: "20"
```
Comment (alexeykazakov) [inline deploy/templates/nstemplatetiers/test/ns_dev.yaml:41]:
```suggestion
      limits.cpu: 20000m
```

File: deploy/templates/nstemplatetiers/test/ns_dev.yaml [hunk 6]
```diff
@@ -13,6 +13,61 @@ objects:
...
+      limits.memory: ${MEMORY_BUILD_LIMIT}
+      requests.cpu: "2"
```
Comment (alexeykazakov) [inline deploy/templates/nstemplatetiers/test/ns_dev.yaml:43]:
```suggestion
      requests.cpu: 2000m
```

File: deploy/templates/nstemplatetiers/test/ns_dev.yaml [hunk 7]
```diff
@@ -13,6 +13,61 @@ objects:
...
+  kind: LimitRange
+  metadata:
+    name: resource-limits
+    namespace: ${USERNAME}-dev
+  spec:
+    limits:
+      - type: "Container"
+        default:
+          cpu: 1000m
+          memory: 3000Mi
```
Comment (alexeykazakov) [inline deploy/templates/nstemplatetiers/test/ns_dev.yaml:66]: You are increasing the default limit by 4 times (750 *4 = 3000) but the overall max limit is being increased by about 2 (7Gi vs 13Gi) plus some extra for builds (10Gi).
Even if we combine terminating and not-terminating 13+10=23Gi it's still fewer containers with the default limits:

7Gi / 750Mi = ~9.3 containers
vs.
23Gi / 3Gi = 7 containers

It means that despite the fact that we are increasing the quota the vast majority of users will actually be able to run fewer pods.
Ideally I would lower the default limit in the limitrange to balance it. 3Gi looks too high. But if you really need so high default limits then I think we have to increase the quota even higher.

Comment (cfchase) [reply deploy/templates/nstemplatetiers/test/ns_dev.yaml:66]: Yes, this was the minimum to get the workshop (3 pods currently running + rollout extra pod + builds) to run without setting requests/limits on those applications.  Unfortunately, the tensorflow model services were at 2500mb, so 3Gi was just a small amount of headroom.  If we want to lower the defaults and just raise the overall quota, I'm fine with that, but the workshops will have to instruct the user to set the limits/requests on those models (which they currently do).

Comment (cfchase) [reply deploy/templates/nstemplatetiers/test/ns_dev.yaml:66]: We are changing our serving methods in a future release in November/December, as well.

File: deploy/templates/nstemplatetiers/test/ns_dev.yaml [hunk 8]
```diff
@@ -54,46 +109,53 @@ objects:
...
+    podSelector: {}
+    policyTypes:
+    - Ingress
 parameters:
 - name: USERNAME
-  required: true
\ No newline at end of file
+  required: true
+- name: MEMORY_LIMIT
+  value: "14Gi"
```
Comment (cfchase) [inline deploy/templates/nstemplatetiers/test/ns_dev.yaml:155]: Did you want these in Mi format as well?

**Comment by cfchase** [comment]
FYI @VedantMahabaleshwarkar @erwangranger this is a change for RHODS workshops to have enough headroom on the dev-sandbox

**Review by alexeykazakov — APPROVED** [review]

**Comment by alexeykazakov** [comment]
/retest

OpenShift CI infra issue

**Pull request merged**